### PR TITLE
Tahoe hawthorn fixes

### PR DESCRIPTION
**Pull request by thraxil**
re-creating this one.

### CODE REVIEW

**Comment by thraxil** [comment]
Squashed a bunch. Still not the prettiest commit log, but better than it was.

File: playbooks/roles/custom_domains/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+edxapp_data_dir: "{{ COMMON_DATA_DIR }}/edxapp"
```
Comment (OmarIthawi) [inline playbooks/roles/custom_domains/defaults/main.yml:2]: Why not include the edxapp default in the `custom_domains` role?

File: playbooks/amc.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@
   roles:
     - { role: swapfile, SWAPFILE_SIZE: "4GB" }
     - role: sudo
-    - custom_domains # This gets the list of domains
+#    - custom_domains # This gets the list of domains
```
Comment (OmarIthawi) [inline playbooks/amc.yml]:
```suggestion
    - role: custom_domains # This gets the list of domains
      when: nginx_enable_custom_domains|default(False)
```

File: playbooks/amc.yml [hunk 3]
```diff
@@ -91,8 +91,8 @@
       nginx_sites:
       - xqueue
     - { role: 'rabbitmq', rabbitmq_ip: '0.0.0.0' }
-    - notifier
-    - { role: notifier, NOTIFIER_DIGEST_TASK_INTERVAL: "5" }
+#    - notifier
```
Comment (OmarIthawi) [inline playbooks/amc.yml]:
```suggestion
```

File: playbooks/amc.yml [hunk 4]
```diff
@@ -91,8 +91,8 @@
...
+#    - { role: notifier, NOTIFIER_DIGEST_TASK_INTERVAL: "5" }
```
Comment (OmarIthawi) [inline playbooks/amc.yml]: Let's get it back? or at least have an open issue in the this repo so we don't loose it.
```suggestion
    - { role: notifier, NOTIFIER_DIGEST_TASK_INTERVAL: "5" }
```

File: playbooks/roles/custom_domains/tasks/main.yml [hunk 5]
```diff
@@ -5,24 +5,33 @@
...
+- name: Fetch custom domains list
   become_user: "{{ edxapp_user }}"
-  django_manage: >
-    command="lms custom_domains_list"
-    app_path="{{ edxapp_code_dir }}"
-    settings="{{ EDXAPP_SETTINGS }}"
-    pythonpath="{{ edxapp_code_dir }}"
-    virtualenv="{{ edxapp_venv_dir }}"
+  shell: >
+    {{ edxapp_venv_bin }}/python manage.py lms --setting={{ EDXAPP_SETTINGS }} custom_domains_list
```
Comment (OmarIthawi) [inline playbooks/roles/custom_domains/tasks/main.yml:20]: I think we need to add `$ source`

Usually it bites at the very wrong time. Having the explicit env enabled is better than just pointing to the correct binary. Same as https://github.com/edx/configuration/pull/4812

```suggestion
    . {{ edxapp_app_dir }}/edxapp_env && {{ edxapp_venv_bin }}/python manage.py lms --setting={{ EDXAPP_SETTINGS }} custom_domains_list
```

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 6]
```diff
@@ -441,6 +441,7 @@
   shell: paver i18n_fastgenerate chdir={{ edxapp_code_dir }}
   sudo_user: "{{ edxapp_user }}"
   environment: "{{ edxapp_environment }}"
+  when: ENABLE_COMPILEMESSAGES
```
Comment (OmarIthawi) [inline playbooks/roles/edxapp/tasks/deploy.yml]:
```suggestion
  when: ENABLE_COMPILEMESSAGES|default(False)
```

File: playbooks/roles/nginx/defaults/main.yml [hunk 7]
```diff
@@ -62,14 +64,6 @@ NGINX_DH_KEYSIZE: 2048
 NGINX_SSL_CERTIFICATE_PATH: '/etc/ssl/certs/{{ NGINX_SSL_CERTIFICATE|basename }}'
 NGINX_SSL_KEY_PATH: '/etc/ssl/private/{{ NGINX_SSL_KEY|basename }}'
 
-# This is only for renaming from the old domain names to the new brand name
-OLD_NGINX_SSL_CERTIFICATE_PATH: '/etc/ssl/certs/old_domain_cert.crt'
-OLD_NGINX_SSL_KEY_PATH: '/etc/ssl/private/old_domain_key.key'
-
-# This is only for renaming from the old domain names to the new brand name
-OLD_NGINX_SSL_CERTIFICATE_CONTENT: ''
-OLD_NGINX_SSL_KEY_CONTENT: ''
```
Comment (OmarIthawi) [inline playbooks/roles/nginx/defaults/main.yml:71]: Finally! Someone brave to remove those

File: playbooks/roles/nginx/tasks/main.yml [hunk 8]
```diff
@@ -296,10 +296,10 @@
     - install:configuration
     - nginx:custom_domains
 
-- name: Reload nginx for custom domains to take effect
-  service: name=nginx state=reloaded
-  when: nginx_enable_custom_domains and inventory_hostname in groups['edx']
-  tags: ['nginx:custom_domains']
+#- name: Reload nginx for custom domains to take effect
```
Comment (OmarIthawi) [inline playbooks/roles/nginx/tasks/main.yml:299]: Let's get this back and set `nginx_enable_custom_domains: False` instead.

**Review by OmarIthawi — APPROVED** [review]
@thraxil a couple of changes then should be good to go.

If you're merging this, please consider adding GitHub issues (or Trello cards) to track the commented out code.

It's just very hard to work with the code when you have a commented out line somewhere buried down in the code.

File: playbooks/roles/custom_domains/defaults/main.yml (hunk 1 above)
Comment (thraxil) [reply playbooks/roles/custom_domains/defaults/main.yml:2]: there's code doing that already: https://github.com/appsembler/configuration/blob/appsembler/ficus/master/playbooks/roles/custom_domains/tasks/main.yml#L3

but it... doesn't work in Hawthorn and I haven't been able to figure out why.

Comment (OmarIthawi) [reply playbooks/roles/custom_domains/defaults/main.yml:2]: We can live with that, until it creates a problem :sweat_smile:

 Thanks @thraxil!

**Pull request merged**